src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -139,19 +140,23 @@ void CalorimeterHitReco::AlgorithmProcess() {
 //        #pragma GCC diagnostic error "-Wsign-converstion"
 
         //did not pass the zero-suppresion threshold
+        const auto cellID = rh->getCellID();
         if (rh->getAmplitude() < m_pedMeanADC + thresholdADC) {
             continue;
         }
 
         // convert ADC to energy
-        const float energy =
-                (((signed) rh->getAmplitude() - (signed) m_pedMeanADC)) / static_cast<float>(m_capADC) * m_dyRangeADC /
+        float energy = (((signed) rh->getAmplitude() - (signed) m_pedMeanADC)) / static_cast<float>(m_capADC) * m_dyRangeADC /
                 m_sampFrac;
-        const float time = rh->getTimeStamp() / stepTDC;
+        if (m_readout == "LFHCALHits" && m_sampFracLayer[0] != 0.){
```

Comment:
What is the point of the `m_sampFracLayer[0] != 0.` check here? Is this a safeguard or for testing without this function?

Reply:
Okay